ml_tools/previewer.py

```diff
@@ -95,74 +113,73 @@ def export_clip_preview(
 
         # increased resolution of video file.
         # videos look much better scaled up
-        if tracker.stats:
-            self.auto_max = tracker.stats["max_temp"]
-            self.auto_min = tracker.stats["min_temp"]
+        if clip.stats:
+            self.auto_max = clip.background_stats["max_temp"]
+            self.auto_min = clip.background_stats["min_temp"]
         else:
             logging.error("Do not have temperatures to use.")
             return
 
         if bool(track_predictions) and self.preview_type == self.PREVIEW_CLASSIFIED:
-            self.create_track_descriptions(tracker, track_predictions)
+            self.create_track_descriptions(clip, track_predictions)
 
-        if self.preview_type == self.PREVIEW_TRACKING and not tracker.frame_buffer.flow:
-            tracker.generate_optical_flow()
+        if self.preview_type == self.PREVIEW_TRACKING and not clip.frame_buffer.flow:
+            clip.generate_optical_flow()
 
         mpeg = MPEGCreator(filename)
 
-        for frame_number, thermal in enumerate(tracker.frame_buffer.thermal):
+        for frame_number, thermal in enumerate(clip.frame_buffer.thermal):
             if self.preview_type == self.PREVIEW_RAW:
                 image = self.convert_and_resize(thermal)
 
             if self.preview_type == self.PREVIEW_TRACKING:
-                image = self.create_four_tracking_image(
-                    tracker.frame_buffer, frame_number
-                )
+                image = self.create_four_tracking_image(clip.frame_buffer, frame_number)
                 image = self.convert_and_resize(image, 3.0, mode=Image.NEAREST)
                 draw = ImageDraw.Draw(image)
-                regions = tracker.region_history[frame_number]
-                self.add_regions(draw, regions)
-                self.add_regions(draw, regions, v_offset=120)
-                self.add_tracks(draw, tracker.tracks, frame_number)
+                # regions = [track.region_bounds for track in  clip.tracks]
+                # regions = tracker.region_history[frame_number]
+                # self.add_regions(draw, regions)
+                # self.add_regions(draw, regions, v_offset=120)
```

Comment:
if these lines aren't needed then delete them please

Reply:
Don't calculate that information anymore. It was showing all regions that were considered. Now we just read the metadata which tells us the tracks.  
But it could show all tracks that were in the metadata? At the moment it just shows the tracks that are being used i.e. have the required confidence. Not sure how important this is?

Reply:
Probably not important